### ISSUE TITLE
electron segfaults on startup

### ISSUE THREAD

**Issue by hoschi** [extr0py/oni#671]
Oni ran fine till the first segfault:
```bash
Aug 28 11:04:35 maxxgubbl kernel: electron[26855]: segfault at 208 ip 00007f2c18421ac5 sp 00007ffe196eeed0 error 4 in libnode.so[7f2c175b1000+12b3000]
Aug 28 11:04:35 maxxgubbl systemd-coredump[26870]: Process 26855 (electron) of user 1000 dumped core.
                                                   
                                                   Stack trace of thread 26855:
                                                   #0  0x00007f2c18421ac5 uv_async_send (libnode.so)
                                                   #1  0x0000000003bb156e _ZN4atom12NodeBindingsD1Ev (electron)
                                                   #2  0x0000000003bb2909 _ZN4atom17NodeBindingsLinuxD0Ev (electron)
                                                   #3  0x0000000003bb9681 _ZN4atom18AtomRendererClientD1Ev (electron)
                                                   #4  0x0000000003bb96c9 _ZN4atom18AtomRendererClientD0Ev (electron)
                                                   #5  0x0000000003b88d09 _ZN4atom16AtomMainDelegateD2Ev (electron)
                                                   #6  0x0000000003b88c3f main (electron)
                                                   #7  0x00007f2c106074ca __libc_start_main (libc.so.6)
                                                   #8  0x000000000059fd79 _start (electron)
```
Then on every start it segfaults ... but I can use it as always. I switched to latest master with same problem. Also I installed latest node LTS version 6.11.2, also no changes. I now updated electron which seems to do the trick:

![2017-08-28-111235_570x322_scrot](https://user-images.githubusercontent.com/163128/29767185-cffdc17e-8be1-11e7-8b3e-ca2c00fcdb06.png)

**Comment by extr0py** [comment]
Oh wow, this is the first electron segfault I've hit (in this project, at least!). Thanks for reporting @hoschi , and letting me know the electron upgrade fixed it.

**Comment by hoschi** [comment]
Seems it isn't that easy :cry: Now the TypeScriptServer won't start anymore:
![2017-08-29-143748_3309x1384_scrot](https://user-images.githubusercontent.com/163128/29821351-dfa3b692-8cc7-11e7-9f6c-8956e25da0f4.png)

**Comment by extr0py** [comment]
Very strange! 

Just a couple questions:
1) What does `process.execPath` return in the console?
2) Does `/home/hoschi/repos/oni-master/node_modules/electron/dist/electron` not exist on your file system?

I would try closing Oni completely (including all electron processes, and reopening). It might be trying to reference the previous electron build

**Comment by hoschi** [comment]
> I would try closing Oni completely (including all electron processes, and reopening). It might be trying to reference the previous electron build

that did the trick :bowing_man:

**Comment by extr0py** [comment]
Excellent! 👍

**Comment by extr0py** [comment]
Opened #679 to track the upgrade blocker (issues with the `makeSingleInstance` API)

**Comment by extr0py** [comment]
Fixed by #695

**Comment by hoschi** [comment]
I moved again to master, works great :+1:
